packages/react-magma-dom/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+
+import { ButtonColor, ButtonSize } from '../Button';
+import { ButtonGroup, ButtonGroupProps } from '../ButtonGroup';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import {
+  setBackgroundColor,
+  ToggleButton,
+  ToggleButtonProps,
+} from '../ToggleButton/ToggleButton';
+import { useGenerateId } from '../../utils';
+
+/**
+ * @children required
+ */
+export interface ToggleButtonGroupProps extends ButtonGroupProps {
+  descriptionId?: string;
+  isInverse?: boolean;
+  /**
+   * Enables a radio configuration throughout the group retaining an active selection at all times.
+   */
+  enforced?: boolean;
+  /**
+   * Sets the Toggle Button group to have only one active selection.
+   */
+  exclusive?: boolean;
+  /**
+   * The onChange handler for managing state of toggle buttons by your custom logic.
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  testId?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+  /**
+   * Value of the toggle button that is the default selected value for the group
+   */
+  value?: string;
+}
+
+export interface ToggleButtonGroupContextInterface {
+  descriptionId?: string;
+  selectedValue?: string;
+  isInverse?: boolean;
+  enforced?: boolean;
+  exclusive?: boolean;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  size?: ButtonSize;
+}
+
+export const ToggleButtonGroupContext =
+  React.createContext<ToggleButtonGroupContextInterface>({});
+
+export const StyledToggleButtonGroup = styled(ButtonGroup)<{
+  descriptionId?: string;
+  onChange?: (event: React.ChangeEvent<HTMLDivElement>) => void;
+}>``;
+
+const StyledWrapper = styled.div<{
+  isInverse?: boolean;
+  enforced?: boolean;
+  selected?: boolean;
+  exclusive?: boolean;
+  size?: ButtonSize;
+  value?: string;
+}>`
+  button {
+    background: ${props => (props.selected ? setBackgroundColor : '')};
+    &:not(:disabled):hover {
+      background: ${props => (props.selected ? setBackgroundColor : '')};
+    }
+    &:not(:disabled):focus {
+      background: ${props => (props.selected ? setBackgroundColor : '')};
+      outline-offset: -2px;
+    }
+    &:hover {
+      background: ${setBackgroundColor};
+    }
+  }
+`;
+
+export const ToggleButtonGroup = React.forwardRef<
+  HTMLDivElement,
+  ToggleButtonGroupProps
+>((props, ref) => {
+  const {
+    children,
+    enforced,
+    exclusive,
+    id: defaultId,
+    isInverse,
+    noSpace,
+    size,
+    value,
+    testId,
+    ...rest
+  } = props;
+  const theme = React.useContext(ThemeContext);
+
+  // Sets the active state for the Toggle Button
+  const [selected, setSelected] = React.useState(null);
+
+  // Sets a specific selected state with the value prop.
+  const [selectedValue, setSelectedValue] = React.useState<string>(value);
```

Comment:
What if the user wants to pre-select multiple buttons? Should this be an array instead of a string?

Reply:
We can check with Orion if this is a requirement or not.